### DISCUSSION TITLE
CTkDataGridView

### DISCUSSION THREAD

**Discussion by RickWalkerOne** [TomSchimansky/CustomTkinter#1601] [Showcase]
## [CTkDataGridView](https://github.com/RickWalkerOne/CTkDataGridView)

If you don't know what is a datagridview, I'll explain. If you already programmed in C# (Windows Forms), you know the datagridview, it's based on they working.

A datagridview is like a table, but different from a table, it works with a datasource, all the changes was did in the datasource, it's works like a list of dicts (rows and columns). If you want to append a new row, you appens in the datasource and automaticaly is changed in the table.

My code is [here](https://github.com/RickWalkerOne/CTkDataGridView), pleasy read the readme.md to know how it works and how implement.

This CTkDataGridView can be updated by user like Entrys and the value is saved in the datasource. The datasource can be traced, and when it happening, you save in the database. If you want, you can made the CTkDataGridView read-only, only to show to you user the data.

The columns have Type validation, so if the type is different, the datagridview will try the convesion.

I made to a implementation with a [CTkMaskedEntry](https://github.com/RickWalkerOne/CTkMaskedEntry/) that I created. So if you want, you can create a mask validation (like a number or date) in all the column.

## how seans:
![screenshot](https://raw.githubusercontent.com/RickWalkerOne/CTkDataGridView/main/Capturar.jpg)

**Comment by avalon60** [comment]
Nice. Another  hidden gem here, is your Mask class. I think that it will also be potentially useful for others, going forward. Thanks for sharing!

**Comment by jklouz** [comment]
wheres CTkDataGridView class?
